webservices/rest.py

```diff
@@ -52,13 +52,8 @@ def sqla_conn_string():
 def limit_remote_addr():
     falses = (False, 'False', 'false', 'f')
     if FEC_API_WHITELIST_IPS not in falses :
-        remote = request.remote_addr
-        route = list(request.access_route)
-        while remote in trusted_proxies:
-            remote = route.pop()
-        print(route.pop())
-
-        if remote not in trusted_proxies:
+        route = request.access_route
+        if route[-2] not in trusted_proxies:
```

Comment:
That's exactly right. (`route[:-2]` is the end-user IP, which we can't whitelist for anyway.) Great suggestion.